### PR TITLE
Add list all Tables API spec and implementation

### PR DESCRIPTION
**Pull request by zhuansunxt**

### CODE REVIEW

**Comment by zhuansunxt** [comment]
@zsxwing Please take a look.

**Comment by zhuansunxt** [comment]
Related issue: https://github.com/delta-io/delta-sharing/issues/64

File: server/src/main/scala/io/delta/sharing/server/SharedTableManager.scala [hunk 1]
```diff
@@ -139,6 +139,28 @@ class SharedTableManager(serverConfig: ServerConfig) {
     }
   }
 
+  def listAllTables(
+      share: String,
+      nextPageToken: Option[String] = None,
+      maxResults: Option[Int] = None): (Seq[Table], Option[String]) = {
+    val shareConfig = getShare(share)
+    val totalSize = shareConfig.schemas.asScala.toList.foldLeft[Int](0) {
```
Comment (zsxwing) [inline server/src/main/scala/io/delta/sharing/server/SharedTableManager.scala:147]: nit: a shorter version: `val totalSize = shareConfig.schemas.asScala.map(_.tables.size).sum`

File: server/src/main/scala/io/delta/sharing/server/SharedTableManager.scala [hunk 2]
```diff
@@ -139,6 +139,28 @@ class SharedTableManager(serverConfig: ServerConfig) {
...
+      (total, s) => s.tables.size + total
+    }
+    getPage(nextPageToken, Some(share), None, maxResults, totalSize) {
+      (start, end) =>
+        shareConfig.schemas.asScala.toList.foldLeft[Seq[Table]](Seq.empty) {
```
Comment (zsxwing) [inline server/src/main/scala/io/delta/sharing/server/SharedTableManager.scala:152]: nit: a shorter version:
```
shareConfig.schemas.asScala.flatMap { schema =>
          schema.tables.asScala.map {
            t =>
              Table(
                name = Some(t.getName),
                schema = Some(schema.name),
                share = Some(share)
              )
          }
        }
```

**Review by zsxwing — COMMENTED** [review]
Could you add end-to-end tests similar to https://github.com/delta-io/delta-sharing/blob/main/server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala#L196-L218 ?

**Comment by zhuansunxt** [comment]
Thanks for the feedback!
I'll update the python client `list_all_tables` in a separate PR

**Review by zsxwing — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**